src/cascadia/TerminalControl/TermControlAutomationPeer.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include <UIAutomationCore.h>
+#include "TermControlAutomationPeer.h"
+#include "TermControl.h"
+#include "TermControlAutomationPeer.g.cpp"
+
+#include "XamlUiaTextRange.h"
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    TermControlAutomationPeer::TermControlAutomationPeer(winrt::Microsoft::Terminal::TerminalControl::implementation::TermControl const& owner) :
+        TermControlAutomationPeerT<TermControlAutomationPeer>(owner),

```

Comment:
This is weird - why is this weird?

Reply:
What part's weird? It's all weird to me haha.

If you're talking about passing in the `owner` but not actually saving it, it's because this is a workaround for getting the RenderData. Realistically, though, I'll probably need the owner to fix the bounding rect issue at some point (but we'll cross that bridge when we get there).

Reply:
Ah. Because we need to pass the Xaml Control to the `FrameworkElementAutomationPeer`. Added a comment.